tests/test_basic_functionality.py

```diff
@@ -237,251 +242,258 @@ def test_tree_creation(empty_tree):
     assert empty_tree.to_dict() == {}
 
 
-def test_tree_coerce_dict():
-    d, s = {}, "test"
-    assert LayeredConfigTree._coerce(d, s) == (d, s)
-    d, s = {"key": "val"}, "test"
-    assert LayeredConfigTree._coerce(d, s) == (d, s)
-    d = {"key1": {"sub_key1": ["val", "val", "val"], "sub_key2": "val"}, "key2": "val"}
-    s = "test"
-    assert LayeredConfigTree._coerce(d, s) == (d, s)
+def test_tree_coerce_dict() -> None:
+    data: NestedDict
+    data, src = {}, "test"
```

Comment:
I think you should just assign to these variables separately:
```suggestion
    data: NestedDict = {}
    src = "test"
```